src/transform/src/dataflow.rs

```diff
@@ -142,17 +146,24 @@ fn inline_views(dataflow: &mut DataflowDesc) {
     }
 }
 
-/// Performs a full optimization pass on the dataflow, including physcal planning
-/// using the supplied set of indexes.
+/// Performs either the logical or the physical optimization pass on the
+/// dataflow using the supplied set of indexes.
 fn optimize_dataflow_relations(
     dataflow: &mut DataflowDesc,
     indexes: &HashMap<GlobalId, Vec<(GlobalId, Vec<MirScalarExpr>)>>,
+    physical: bool,
```

Comment:
I think I'd rather pass the optimizer instance here. Also, now that `for_dataflow` no longer contains logical optimizations, we could just rename them as `logical_optimizer` and `physical_optimizer`.